packages/hl7v2-server/src/app.ts

```diff
@@ -0,0 +1,47 @@
+import { Hl7Server } from "@medplum/hl7";
+import * as dotenv from "dotenv";
+import { ConsoleLogger, Logger } from "./logger";
+dotenv.config();
+
+const HL7_PORT = process.env.HL7_PORT ? parseInt(process.env.HL7_PORT) : 2575;
+
+async function createHl7Server(logger: Logger): Promise<Hl7Server> {
+  const server = new Hl7Server(connection => {
+    logger.info("Connection received");
+
+    connection.addEventListener("message", ({ message }) => {
+      logger.debug(
+        `## New Message from ${connection.socket.remoteAddress}:${connection.socket.remotePort} ##`
+      );
+      message.segments.forEach(segment => {
+        logger.debug("Segment:", segment.toString());
+      });
+      logger.debug("\n\n");
+      connection.send(message.buildAck());
+    });
```

Comment:
Here that will send messages to SQS, right?

Also, we could prob acknowledge the message right at the beginning so while we I/O to log and SQS the response is on its way to the client?

Reply:
Hm I thought for lower likelihood to fail we would want to ack only _after_ the message has made it on to the queue, and has thus been persisted. if we ack beforehand there's a chance we drop the message. 

if we don't ack the message, the HL7 client is supposed to keep retrying until we do. Acking after msg makes it onto the queue seems good to ensure we don't drop messages.

We actually also need to explicitly NACK misformatted messages and messages that we're not going to be able to process. I'll add that to my MLLP specific todo list

https://www.notion.so/metriport/MLLP-Proj-Notes-1a0b51bb904080909d99d7b33ad4228a